src/hooks/core/chat/useDMMessage.ts

```diff
@@ -0,0 +1,145 @@
+      }
+
+      const {
+        anonUserInfoEmojiName,
+        anonUserInfoColorCode,
+        anonUserInfoColorName,
+        anonUserInfoEmojiCode,
+        channelName,
+        channelImage
+      } = getChannelListInfo(builtChannelData, signedProfileId, anonProfileId);
+
+      const chatType = type === 'ANONYMOUS' ? 'ANON_PM' : 'PM';
+      const chatName = {
+        name: channelName,
+        image: channelImage
+      };
+
+      channel.firstMessage = channel?.messages?.[channel?.messages?.length - 1];
+      channel.myUserId = signedProfileId; // change to use getChannelListInfo
+      channel.targetName = chatName?.name; // change to use getChannelListInfo
+      channel.targetImage = chatName?.image; // change to use getChannelListInfo
+
+      channel.channel = {...channel};
+      console.log('data', {
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      });
+      const channelList = ChannelList.fromChannelAPI(channel, chatType, channel?.members, {
+        anon_user_info_color_name: anonUserInfoColorName,
+        anon_user_info_color_code: anonUserInfoColorCode,
+        anon_user_info_emoji_code: anonUserInfoEmojiCode,
+        anon_user_info_emoji_name: anonUserInfoEmojiName
+      });
+      await channelList.saveIfLatest(localDb);
+      refresh('channelList', 'chat', 'channelInfo', 'channelMember');
+      return goToChatScreen(channelList);
+    } catch (error) {
+      console.log('sendMessage', error);
+    }
+  };
+
+  const sendMessageDM = async (
+    id,
+    source: 'post' | 'comment',
+    channelCategory: ChannelCategory
+  ) => {
+    const initChat =
+      channelCategory === 'ANONYMOUS'
+        ? await initChatFromPostAnon({source, id})
+        : await initChatFromPost({source, id});
+
+    const builtChannelData = {
+      better_channel_member: initChat.data.better_channel_members,
+      members: initChat?.data.members
+    };
+
+    const channel = {
+      ...initChat?.data?.channel,
+      better_channel_members: initChat?.data?.better_channel_members,
+      members: initChat?.data?.better_channel_members
+    };
+
+    if (!channel?.members?.length) {
+      console.info('no members');
+      return;
+    }
+
+    if (channel?.type === 'topics') {
+      console.info('type is topics');
+      return;
+    }
+
+    try {
+      await Promise.all(
+        (channel?.members || []).map(async (member) => {
+          const userMember = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+          const memberSchema = ChannelListMemberSchema.fromWebsocketObject(
+            channel?.id,
+            uuid(),
+            member // change to use getChannelListInfo
+          );
+          await userMember.saveOrUpdateIfExists(localDb);
+          await memberSchema.save(localDb);
+        })
+      );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Here you're using `Promise.all` to perform multiple asynchronous operations concurrently. This is good for performance, but keep in mind that `Promise.all` fails fast, meaning if one promise rejects, it will reject immediately with the reason of the first promise that rejected. If you want to ensure all operations are completed even if some fail, consider using `Promise.allSettled` instead.

60:
This line seems to be creating a new property `channel` on the `channel` object itself which could lead to confusion. Consider renaming this property to avoid confusion.
```diff
-      channel.channel = {...channel};
+      channel.channelData = {...channel};
```

Overall, the code is well structured and follows good practices for hooks in React. The use of async/await syntax makes the asynchronous code easier to read and understand. The separation of concerns into different functions (`saveChannelList`, `sendMessageDM`) also improves readability and maintainability.


<!-- This is an auto-generated comment by OSS CodeRabbit -->